### DISCUSSION TITLE
Challenge 9 help needed!! (Spoilers!!!!)

### DISCUSSION THREAD

**Discussion by Abelaby** [Cyfrin/foundry-full-course-cu#332] [Q&A]
Hey, I cant seem to get the correct answer for challenge 9:
I tried deploying a contract to Sepolia to check it out and the answer Im getting is wrong.
I tried different helper functions but none of them seem to work:
Here is my test contract:

```solidity

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract TestBlock{
    function getBlockNumber()public view returns (uint256){
        return block.timestamp;
    }
    function getPervando() public view returns(uint256){
        return block.prevrandao;
    }
    function getAnswer()public view returns(uint256 ){
        uint256(keccak256(abi.encodePacked(msg.sender, block.prevrandao, block.timestamp))) % 100000;
    }
    function OwnAnswer(address myAddress) public  view returns (uint256){
         uint256(keccak256(abi.encodePacked(myAddress, block.prevrandao, block.timestamp))) % 100000;
    }
    function getSender() public view returns(address){
        return msg.sender;
    }
    function getAbiEnocoded() public view returns (bytes memory){
        return (abi.encodePacked(msg.sender, block.prevrandao, block.timestamp));
    }
    function getKeccack()public view returns (bytes32){
        return (keccak256(abi.encodePacked(msg.sender, block.prevrandao, block.timestamp)));
    }
    function getUintValue() public view returns (uint256){
        return  uint256(keccak256(abi.encodePacked(msg.sender, block.prevrandao, block.timestamp)));
    }
}
```

Get UintValue() actually returns me  a large value but when I call getAnswer() im always getting zero:

Can someone help me?

**Comment by ghost** [comment]
Three things:

1. You're getting zero because you're missing a couple of `return` statements.
2. You're getting a huge number from `getUintValue` because you're forgot to include something.
3. How are you using this contract to solve the challenge?

**Reply by cromewar** [reply]
Nice approach @pacelliv we do ask for you not to spoil the challenges by giving the solutions, so thanks for keeping this this way.
@Abelaby have you being able to solve it?

**Reply by ghost** [reply]
Don't worry @cromewar, I promise I will not spill the beans.

**Comment by Abelaby** [comment]
Hey @cromewar  and @pacelliv  thanks for your kind recommendations, but I havent find the solution to the challenge.

Im getting some values but those doesnt seem to work.
I played around with the script by removing the block.prevrandao() and even tried by replacing it by block.difficulty.

But I cant seem to get it right

**Reply by ghost** [reply]
You're starting to deviate, you were on a good path. Don't dwell too much on it, you can come back.

**Reply by Abelaby** [reply]
I will come back to this later, Im currently unable to solve challenge 8 and 9.
I got the nft for challenge 10 though, maybe I will try these two challenges after I complete lesson 11.

I will update you guys later on my progress, thanks guys

**Reply by Abelaby** [reply]
Hey guys, Im back.
I have a question, since the variables for the random number are ```block.timestamp ``` and ```block.prevrandao``` so does it mean that the values will be different from the contract deployed by me and our challenge contract or am I wrong, its getting a lil confusing.

**Reply by ghost** [reply]
The [Block Properties](https://docs.soliditylang.org/en/latest/cheatsheet.html#block-and-transaction-properties) always return current values.

> values will be different from the contract deployed by me and our challenge contract or am I wrong, its getting a lil confusing

If you completed the first four lessons I'm sure you can think how to test this.

**Reply by Abelaby** [reply]
But ``` block.prevrandao ``` only seems to work in paris EVM ?

I dont see a paris evm in my remix

**Reply by ghost** [reply]
I don't think you need to select it, Remix should select that version by default. In my case it shows the evm version above the deploy button after connecting to Sepolia.

**Reply by basefee** [reply]
@pacelliv Need a small hint, how to call the challenge contract with the value obtained in the getAnswer function?

**Reply by sharonphiliplima** [reply]
Guys, are you using tx.orign or address(this)?
